source/module_hamilt_lcao/hamilt_lcaodft/LCAO_gen_fixedH.cpp

```diff
@@ -289,9 +289,28 @@ void LCAO_gen_fixedH::build_ST_new(const char& dtype, const bool& calc_deri, con
 								{
 									if(dtype=='S')
 									{
-										this->LM->DSloc_Rx[nnr] = olm[0];
-										this->LM->DSloc_Ry[nnr] = olm[1];
-										this->LM->DSloc_Rz[nnr] = olm[2];
+										if (GlobalV::NSPIN != 4)
```

Comment:
define nspin before the loop, pass GlobalV::NSPIN value to nspin, and change this sentence to "if(nspin==1 || nspin==2)"